src/main/java/sparksoniq/spark/iterator/flowr/ForClauseSparkIterator.java

```diff
@@ -314,4 +320,30 @@ public void print(StringBuffer buffer, int indent)
         buffer.append("\n");
         _expression.print(buffer, indent+1);
     }
+    
+    public Map<String, DynamicContext.VariableDependency> getProjection(Map<String, DynamicContext.VariableDependency> parentProjection)
+    {
+        if(_child == null)
+        {
+            return null;
+        }
+
+        // passing dependencies to parent
```

Comment:
I initially found this comment to be a bit confusing. I believe this refers to line 335 and following lines?

I would suggest, if possible, extracting such a dependency calculation inside the getProjection function into a separate function, rather than relying on a comment

Reply:
Good catch. This was a leftover before I cleaned up the API.